ts/torch_handler/distributed/base_neuronx_continuous_batching_handler.py

```diff
@@ -0,0 +1,411 @@
+                + str(tp_degree)
+                + " are not available: "
+                + str(error)
+            )
+
+            raise error
+        self._set_class(ctx)
+        self.tokenizer = self.tokenizer_class.from_pretrained(
+            model_checkpoint_path, return_tensors="pt", padding_side="left"
+        )
+        self.tokenizer.pad_token = self.tokenizer.eos_token
+        self.tokenizer.pad_token_id = self.tokenizer.eos_token_id
+        self.tokenizer.padding_side = "left"
+
+        continuous_batching_config = ContinuousBatchingConfig(
+            batch_size_for_shared_caches=self.batch_size
+        )
+        neuron_config = NeuronConfig(continuous_batching=continuous_batching_config)
+        kwargs = dict(
+            tp_degree=tp_degree,
+            amp=amp,
+            batch_size=self.batch_size,
+            n_positions=[self.max_length],
+            context_length_estimate=ctx.model_yaml_config.get("handler", {}).get(
+                "context_length_estimate", [self.max_length]
+            ),
+            neuron_config=neuron_config,
+        )
+        self.model = self.model_class.from_pretrained(model_checkpoint_path, **kwargs)
+        logger.info("Starting to compile the model")
+        self.model.to_neuron()
+        logger.info("Model has been successfully compiled")
+
+        # 1D: [seq_id]
+        # an empty slot if seq_id is -1, otherwise 0
+        self.decode_seq_ids = torch.full([self.batch_size], -1)
+        # 2D:[batch_size, next_cache_id]
+        self.decode_cache_ids = torch.zeros(self.batch_size, 1, dtype=torch.int64)
+        # 2D: [batch_size, next_token]
+        self.decode_next_tokens = torch.zeros(self.batch_size, 1, dtype=torch.int64)
+        # self.decode_next_tokens = torch.full(self.batch_size, self.tokenizer.eos_token_id)
+
+        for _, seq_id in enumerate(reversed(range(self.batch_size))):
+            self.empty_seq_ids.append(seq_id)
+
+        logger.info("Model %s loaded successfully", ctx.model_name)
+        self.initialized = True
+
+    def preprocess(self, requests):
+        prefill_req_ids, prefill_seq_ids, prefill_input_text, req_decode_seq_ids = (
+            [],
+            [],
+            [],
+            [],
+        )
+        for req_id, req_data in zip(self.context.request_ids.values(), requests):
+            if req_id not in self.context.cache:
+                prefill_req_ids.append(req_id)
+                seq_id = self._get_empty_seq_id()
+                self.seq_id_to_req_id[seq_id] = req_id
+                prefill_seq_ids.append(seq_id)
+
+                data = req_data.get("data") or req_data.get("body")
+                if isinstance(data, (bytes, bytearray)):
+                    data = data.decode("utf-8")
+
+                prompt = data.get("prompt")
+                max_new_tokens = int(data.get("max_new_tokens", self.max_new_tokens))
+                prefill_input_text.append(prompt.strip())
```

Comment:
Better to not strip as the whitespace could have meaning and has influence the model return.